xcube/util/jsonschema.py

```diff
@@ -101,10 +101,26 @@ def validate_instance(self, instance: Any):
         # jsconschema needs extra packages installed to validate some formats;
         # if they are missing, the format check will be skipped silently. For
         # date-time format, strict_rfc3339 or rfc3339-validator is required.
+
+        schema = self.to_dict()
+
+        # By default, jsonschema only recognizes lists as arrays. Here we derive
+        # and use a custom validator which recognizes both lists and tuples as
+        # arrays.
+        base_validator = jsonschema.validators.validator_for(schema)
+        new_type_checker = \
+            base_validator.TYPE_CHECKER.redefine(
+                'array',
+                lambda checker, inst: isinstance(inst, (list, tuple)))
+        custom_validator = \
+            jsonschema.validators.extend(base_validator,
+                                         type_checker=new_type_checker)
+
         jsonschema.validate(instance=instance, schema=self.to_dict(),
```

Comment:
```suggestion
        jsonschema.validate(instance=instance, schema=schema,
```

otherwise `schema.to_dict()` is computed twice

Reply:
Assuming you mean self, this will lead to a lot of test failures.

Reply:
I'm probably missing something here, but I can't find the other invocation of `to_dict()`, `schema` isn't defined, and if `schema` is a typo for `self` I don't understand how jsonschema is supposed to validate a `JsonSchema` object rather than the dict it's expecting.